src/main/java/lotto/type/SingleNumber.java

```diff
@@ -0,0 +1,83 @@
+package lotto.type;
+
+/**
+ * 로또 번호나 당첨 번호에 사용되는 번호 1개를 나타냅니다.
+ * 불변 객체입니다.
+ */
+public class SingleNumber implements Comparable<SingleNumber> {
+
+    /** 로또 번호 하한선 (경계값 포함) */
+    private static final int MIN_NUM = 1;
+
+    /** 로또 번호 상한선 (경계값 포함) */
+    private static final int MAX_NUM = 45;
+
+    private int lottoNumber;
+
+    private SingleNumber() {
+    }
+
+    public static SingleNumber of(int lottoNumber) {
+        checkRange(lottoNumber);
+
+        SingleNumber lotto = new SingleNumber();
+        lotto.lottoNumber = lottoNumber;
+
+        return lotto;
```

Comment:
```suggestion
        return new SingleNumber(lottoNumber);
```

생성 후 설정해주기보다는 한 번에 완전한 불변 클래스를 만들어주는 방법에 대해서도 고민해보세요.
어떤 장점이 있을까요?